### PR TITLE
add MaxBalance Module

### PR DESCRIPTION
**Pull request by Joachim-Lebrun**
- [x] implement MaxBalance Module
- [x] Test MaxBalance Module for full module coverage
- [x] Add Documentation for MaxBalance Module on docs.tokeny.com

### CODE REVIEW

File: contracts/compliance/modular/modules/MaxBalanceModule.sol [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        _IDBalance[msg.sender][_idTo] += _value;
+        if (_IDBalance[msg.sender][_idTo] > _maxBalance[msg.sender]) revert MaxBalanceExceeded(msg.sender, _value);
+    }
+
+    /**
+     *  @dev See {IModule-moduleBurnAction}.
+     *  no burn action required in this module
+     */
+    function moduleBurnAction(address _from, uint256 _value) external override onlyComplianceCall {
+        address _idFrom = _getIdentity(msg.sender, _from);
```
Comment (Nakasar) [inline contracts/compliance/modular/modules/MaxBalanceModule.sol:152]: You might want to check if the `_idFrom` is the null address (I think in our case here `_getIdentity()` returns the zero address if the identity was not found).

File: contracts/compliance/modular/modules/MaxBalanceModule.sol [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        _IDBalance[msg.sender][_idFrom] -= _value;
+        if (_IDBalance[msg.sender][_idTo] > _maxBalance[msg.sender]) revert MaxBalanceExceeded(msg.sender, _value);
+    }
+
+    /**
+     *  @dev See {IModule-moduleMintAction}.
+     *  no mint action required in this module
+     */
+    function moduleMintAction(address _to, uint256 _value) external override onlyComplianceCall {
+        address _idTo = _getIdentity(msg.sender, _to);
```
Comment (Nakasar) [inline contracts/compliance/modular/modules/MaxBalanceModule.sol:142]: Same, _idFrom can be the zero address I think?

File: contracts/compliance/modular/modules/MaxBalanceModule.sol [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        }
+    }
+
+    /**
+     *  @dev See {IModule-moduleTransferAction}.
+     *  no transfer action required in this module
+     */
+    function moduleTransferAction(address _from, address _to, uint256 _value) external override onlyComplianceCall {
+        address _idFrom = _getIdentity(msg.sender, _from);
+        address _idTo = _getIdentity(msg.sender, _to);
```
Comment (Nakasar) [inline contracts/compliance/modular/modules/MaxBalanceModule.sol:131]: Same, _idTo can be the zero address I think?

File: contracts/compliance/modular/modules/MaxBalanceModule.sol (hunk 1 above)
Comment (aliarbak) [reply contracts/compliance/modular/modules/MaxBalanceModule.sol:152]: done

**Review by Nakasar — APPROVED** [review]

**Pull request merged**